### ISSUE TITLE
Add json log format flag for Workflow controller

### ISSUE THREAD

**Issue by paragones92** [argoproj/argo-workflows#7252]
# Summary

Just as other components of Argo workflows Server and ArgoCD have the option to set the log format of json or text, add this option as well to argo workflow controller.

```
      --logformat string                Set the logging format. One of: text|json (default "text")
```

# Use Cases

It would be easier for logs scrapper like Fluentd or Fluentbit to use the json parsers to better parse the logs to the desired logs storage system.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

**Comment by terrytangyuan** [comment]
This has been supported in https://github.com/argoproj/argo-workflows/pull/5158 unless I misunderstood the intention.

You can use `--log-format` when you start the workflow controller.

**Comment by paragones92** [comment]
Oh thanks didnt find that in the code and i thought it was not implemented. Thank you!

**Comment by DrissiReda** [comment]
Actually even when I use this, I have different log formats:

```json
{"level":"info","msg":"Update leases 200","time":"2022-03-08T15:36:38.213Z"}
{"level":"info","msg":"Get leases 200","time":"2022-03-08T15:36:43.218Z"}
{"level":"info","msg":"Update leases 200","time":"2022-03-08T15:36:43.222Z"}
{"level":"info","msg":"List workflowtasksets 404","time":"2022-03-08T15:36:47.671Z"}
E0308 15:36:47.672055 1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/reflector.go:167: Failed to watch *v1alpha1.WorkflowTaskSet: failed to list *v1alpha1.WorkflowTaskSet: the server could not find the requested resource (get workflowtasksets.argoproj.io)
{"level":"info","msg":"Get leases 200","time":"2022-03-08T15:36:48.251Z"}
{"level":"info","msg":"Update leases 200","time":"2022-03-08T15:36:48.258Z"}
{"level":"info","msg":"Watch pods 200","time":"2022-03-08T15:36:51.718Z"}
```

This leads me to believe there are multiple log outputs.

**Comment by cobyrne09** [comment]
I believe I'm seeing the same issue... most errors are being logged in json, but still getting some such as `W0707 00:22:13.789315       1 warnings.go:70] external-secrets.io/v1alpha1 ExternalSecret is deprecated; use external-secrets.io/v1beta1 ExternalSecret` coming through. Guessing it's just a matter of different processes outputting differently... will dig deeper if I get the chance.

**Comment by KimPaow** [comment]
Any updates on this?

**Comment by agilgur5** [comment]
> This leads me to believe there are multiple log outputs.

Indeed that is the log output from [`kubernetes/client-go`](https://github.com/kubernetes/client-go), which does not have a JSON format option afaik. If you know how to configure that, please submit a PR or file a new issue about that.

Otherwise locking this as resolved.

> Any updates on this?

This is an issue that was closed 2.5 years ago with a last comment 2 years ago. There is nothing to update on.

And even if there was, [any updates](https://justinmayer.com/posts/any-updates/) [would be in the thread](https://sindresorhus.com/blog/issue-bumping)
